hypothesis-python/src/hypothesis/internal/filtering.py

```diff
@@ -86,27 +90,40 @@ def comp_to_kwargs(x: ast.AST, op: ast.AST, y: ast.AST, *, argname: str) -> dict
         # (and we can't even do `arg == arg`, because what if it's NaN?)
         raise ValueError("Can't analyse this comparison")
 
+    kwargs: Dict[str, Union[Any, bool]] = {}
+    if isinstance(x, ast.Call) and isinstance(x.func, ast.Name) and x.func.id == "len":
+        kwargs["len_func"] = True
+    if isinstance(y, ast.Call) and isinstance(y.func, ast.Name) and y.func.id == "len":
+        kwargs["len_func"] = True
```

Comment:
```suggestion
    of_len = {"len": True} if isinstance(x, ast.Call) or isinstance(y, ast.Call) else {}
```

Then below, we can `return {..., **of_len}` instead of mutating kwargs.  I quite strongly prefer this style because it's easy to see at a glance that each branch must be independent, because each branch immediately returns - and so there are few opportunities for bugs to sneak in.  https://grugbrain.dev expresses my feelings on this well.